src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -910,12 +910,29 @@ private static async Task<DurableOrchestrationStatus> GetDurableOrchestrationSta
                         }
                     }
 
-                    var counter = 0;
-                    indexList.Sort();
-                    foreach (var indexValue in indexList)
+                    if (eventTypes != null)
                     {
-                        historyArray.RemoveAt(indexValue - counter);
-                        counter++;
+                        for (int i = historyArray.Count - 1; i >= 0; i--)
+                        {
+                            var eventTypeName = historyArray[i].Value<string>("EventType");
+                            if (Enum.TryParse(eventTypeName, out EventType eventType))
+                            {
+                                if (!eventTypes.Contains(eventType))
+                                {
+                                    historyArray.Remove(historyArray[i]);
+                                }
+                            }
+                        }
+                    }
+                    else
+                    {
+                        var counter = 0;
+                        indexList.Sort();
+                        foreach (var indexValue in indexList)
+                        {
+                            historyArray.RemoveAt(indexValue - counter);
+                            counter++;
+                        }
```

Comment:
Can you please help me understand what this code does? it seems this would always execute before, so I'm trying to understand why we're not always running this logic anymore

Reply:
This part is also to edit the historyAarray list. In the original code, this part will delete some eventTypes and only leave several importane ones left to make the historyArray list more readable. Because we have more than 10 eventTypes and the non-edited list is quite long. So, if the caller wants to query other eventTypes, it will go to if condition. Otherwise, it will go to this else part, which is the original one, return only certain eventTypes.